### PR TITLE
Change logic, add error message, add tests

### PR DESCRIPTION
**Pull request by amoodie**
I pulled your feature branch down to make the tests, and found some bugs with how the parser was working. I believe I have corrected them here.

I guess the major change I should point out is that I set it up to throw errors if 1) an `input_file` is specified and not found, and 2) if an incorrect type was given. Before, 1) defaults were used and the run proceeded, and 2) a message was printed and the default was used. I found this to be confusing behavior.

In the process, I reordered and changed some of your logic in the parser. It still functions the same, but I think it is more readable and simpler. I also changed the `**kwargs` input to `pyDeltaRCM` to be a defined keyword (default `None`), and then used it as a sort-of boolean in the `init_tools` flow you set up. 

Additionally, I have implemented a series of tests that check the workflow of the input parser, and have verified that it works as expected (at least how I expect it to...)
 
Additionally, I updated the documentation.

### CODE REVIEW

**Comment by amoodie** [comment]
tests here: https://travis-ci.com/github/amoodie/pyDeltaRCM_WMT/builds/161523478

**Comment by amoodie** [comment]
My only thought is whether a better name for `input_file` might be `config_file` or just `config`. Not too concerned about this though.

File: pyDeltaRCM/init_tools.py [hunk 1]
```diff
@@ -48,39 +48,40 @@ def import_files(self):
...
+                raise e
         else:
-            print('The specified input file could not be found. Using default values...')
             user_dict = dict()
 
         # go through and populate input vars with user and default values,
         # checking user values for correct type.
-
         for oo in default_dict.keys():
-            the_type = eval(default_dict[oo]['type'])
```
Comment (ericbarefoot) [inline pyDeltaRCM/init_tools.py:72]: This was a bad variable name, yes.

**Review by ericbarefoot — APPROVED** [review]
This looks good to me. I tend to prefer to put ` = ` and keep the spaces, since it looks less cramped to me. But otherwise I like it. I didn't look too hard at the tests, so I'll merge this and pull it down, and see if it breaks anything for me.

**Pull request merged**